src/cljs/witan/ui/controllers/datastore.cljs

```diff
@@ -37,10 +37,15 @@
 
 (defn add-new-metadata-to-app-state!
   [md]
-  (let [new-datapack-meta (assoc-in md
-                                    [:kixi.datastore.metadatastore/provenance :kixi/user] (data/get-in-app-state :app/user))]
-    (data/swap-app-state! :app/data-dash update :items #(cons new-datapack-meta %))
-    (data/swap-app-state! :app/datastore update :ds/file-metadata #(assoc % (:kixi.datastore.metadatastore/id new-datapack-meta) new-datapack-meta))))
+  (let [new-id (:kixi.datastore.metadatastore/id md)
+        new-meta (assoc-in md
+                           [:kixi.datastore.metadatastore/provenance :kixi/user] (data/get-user))
```

Comment:
how come this isn't already present?

Reply:
Because we don't send the user as part of the provenance, it's added by the datastore. This add function is transforming to make it look as if it's come from datastore.